### PR TITLE
Selection action

### PR DESCRIPTION
**Pull request by jrhe**
Add an action which gets fired when the selection changes with the current selection. 
May want to deprecate two way selection binding to adhere more to the Data down, actions up pattern.

### CODE REVIEW

File: addon/components/ember-table.js [hunk 1]
```diff
@@ -105,6 +105,10 @@ StyleBindingsMixin, ResizeHandlerMixin, {
     }
   }).property('persistedSelection.[]', 'rangeSelection.[]', 'selectionMode'),
 
+  _selectionChanged: Ember.observer('selection', function() {
```
Comment (azirbel) [inline addon/components/ember-table.js]: should observe `selection.[]`, right?

Comment (jrhe) [inline addon/components/ember-table.js]: Correct. No idea why/ how this is working in  my app.

**Comment by azirbel** [comment]
I'm in favor of adding this action, but (1) I think it should be called `onSelectionChanged` to be consistent with https://github.com/Addepar/ember-table/pull/350, and (2) I'm not convinced using an observer is the right way to do it. This means e.g. if you use the two-way binding to set `selection` from outside the table, the observer will fire. But the action should only be triggered when table-internal logic makes the selection change. Maybe this `sendAction` call belongs in `setSelected` (and may have to be added elsewhere as well).

**Comment by billy-addepar** [comment]
This version of Ember table is no longer supported. If you want to continue discussion, you can open the issue on [Ember Table Legacy](https://github.com/Addepar/ember-table-legacy)

**Pull request closed without merging**